src/main.rs

```diff
@@ -0,0 +1,169 @@
+        // load data for the first
+        load_data(connection.clone()).await?;
+        println!("\nSchema Loaded\n");
+    }else {
+        println!("\nSchema Already Defined\n");
+    }
+
+    Ok(())
+}
+
+async fn query1(connection: Connection,q: String)->std::io::Result<()>{
+    let databases = DatabaseManager::new(connection.clone());
+    let session = Session::new(databases.get(TEST_DATABASE).await.unwrap(), Data).await.unwrap();
+    let transaction = session.transaction(Read).await.unwrap();
+    println!("::Q1::");
+    let mut answer_stream = transaction.query().match_(q.as_str()).unwrap();
+    while let Some(result) = answer_stream.next().await{
+        match result {
+            Ok(concept_map) => {
+                for (_, concept) in concept_map {
+                    if let Concept::Attribute(Attribute { value: Value::String(value), .. }) = concept {
+                        println!("{}",value);
+                    }
+                }
+            }
+            Err(err) => {
+                panic!("An error occurred fetching answers of a Match query: {err}")
+            }
+        }
+    }
+    Ok(())
+}
+async fn query2(connection: Connection,q: String)->std::io::Result<()>{    
+    let databases = DatabaseManager::new(connection.clone());
+    let session = Session::new(databases.get(TEST_DATABASE).await.unwrap(), Data).await.unwrap();
+    println!("::Q2::");
+    let transaction = session.transaction(Read).await.unwrap();
+    let answer = transaction.query().match_aggregate(q.as_str()).await.unwrap();
+    println!("Answe for Q2:  {}\n",answer.into_i64());
+    Ok(())
+}
+async fn query3(connection: Connection,q: String)->std::io::Result<()>{    
+    let databases = DatabaseManager::new(connection.clone());
+    let session = Session::new(databases.get(TEST_DATABASE).await.unwrap(), Data).await.unwrap();
+    let transaction = session.transaction(Read).await.unwrap();
+    println!("::Q3::");
+    let mut answer_stream = transaction.query().match_(q.as_str()).unwrap();
+    while let Some(result) = answer_stream.next().await{
+        match result {
+            Ok(concept_map) => {
+                for (_, concept) in concept_map {
+                    if let Concept::Attribute(Attribute { value: Value::String(value), .. }) = concept {
+                        println!("{}",value);
+                    }
+                }
+            }
+            Err(err) => {
+                panic!("An error occurred fetching answers of a Match query: {err}")
+            }
+        }
+    }
+    Ok(())
+}
+
+async fn runIO(connection: Connection){
+    println!("\nQ1->What places could buy raw_food=$x ?\nQ2->Get count of non-vegetarian outlets with vegetarian specialities.\nQ3->Get count raw items sold at places with avg_rating more tha $r and has a dish using it as raw_ingredient with price greater than $p units.\nWhat query would you like to make? Enter 1,2 or 3.\n");
+    let qtype: i32=read!();
+    if qtype==1{
```

Comment:
We're using this `if qtype == x` style of determining which query to run. Instead, we could use a match statement. Something like
```rs
match qtype {
    1 => query1(connection, f).await.unwrap(),
    2 => query2(connection, f).await.unwrap(),
    3 => query3(connection, f).await.unwrap(),
    _ => println!("Query entered is not 1,2 or 3\n")
};
```
would be more ideal, as it is easier to read and less error prone if we were to expand this in the future.